### PR TITLE
Subcircuit layout upgrade #48

### LINKED ISSUE #36: SubCircuit Surface LED's
When customizing the layout of a SubCircuit, any LED's that have been placed should appear on the SubCircuit similar to inputs or outputs, but on the surface. A similar idea I just had is to do the same with buttons, allowing for a SubCircuit button panel. 
Here's a graphical representation of what I mean about the surface LED's:
screenshot 2

### PR DESCRIPTION
**Pull request by SubhajitCode**
Fixes #36 

#### Describe the changes you have made in this pr -

implements the the surface led (circular can be changed to rectangular..)
label is not visible yet and the layout size some time unstable ...
need more thorough work  and guidance from the core team ...
check before merging because all leds will be surface leds for now needed to work on that also
I request to be assigned on this issue for GSOC 2019

### Screenshots of the changes (If any) -

toggling led still pic

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,7 +7,11 @@
 # Ignore bundler config.
 /.bundle
 /.idea
-
+#my ignore specific to my environment
+/public/js/*.json
+/docker-compose.yml
+/Dockerfile
+#end ofmy ignore specific to my environment
```
Comment (satu0king) [inline .gitignore]: /docker-compose.yml and /Dockerfile cannot be in .gitignore!

File: public/js/logix.js [hunk 2]
```diff
@@ -39,14 +39,14 @@ forceResetNodes = true; // FLag to reset all Nodes
 circuitElementList = [
     "Input", "Output", "NotGate", "OrGate", "AndGate", "NorGate", "NandGate", "XorGate", "XnorGate", "SevenSegDisplay", "HexDisplay",
     "Multiplexer", "BitSelector", "Splitter", "Power", "Ground", "ConstantVal", "ControlledInverter", "TriState", "Adder", "Rom", "RAM", "TflipFlop",
-    "JKflipFlop", "SRflipFlop", "DflipFlop", "TTY", "Keyboard", "Clock", "DigitalLed", "Stepper", "VariableLed", "RGBLed", "SquareRGBLed", "Button", "Demultiplexer",
+    "JKflipFlop", "SRflipFlop", "DflipFlop", "TTY", "Keyboard", "Clock", "Stepper", "VariableLed", "RGBLed", "SquareRGBLed", "Button", "Demultiplexer",
```
Comment (satu0king) [inline public/js/logix.js]: You cannot just remove "DigitalLed from here!'

**Comment by satu0king** [comment]
This looks like a good start and you are right about more thorough work and guidance from the core team. This is quite a critical module and we would want it to be done only during the GSoC period. You should try and tackle easier issues like #155 for now.

File: public/js/logix.js (hunk 2 above)
Comment (jbox144) [reply public/js/logix.js]: He can, and he did. Not that he should have... but he did.

Comment (SubhajitCode) [reply public/js/logix.js]: thanks for the feedback I will correct the mistake and create another pull req and definitely look into issue #115

**Comment by satu0king** [comment]
As previously discussed, this is a big project, and we would love that you tackle this as a GSoC project. You are of course open to experiments and write your findings in the proposal. We can then discuss which is the best approach based on your thoughts.

**Pull request closed without merging**